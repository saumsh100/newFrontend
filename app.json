  {
  "name": "carecru",
  "scripts": {
  },
  "env": {
    "NODE_ENV": {
      "required": true
    },
    "RETHINKDB_AUTHKEY": {
      "required": true
    },
    "RETHINKDB_DB": {
      "required": true
    },
    "RETHINKDB_HOST": {
      "required": true
    },
    "RETHINKDB_PORT": {
      "required": true
    },
    "VENDASTA_API_KEY": {
      "required": true
    },
    "VENDASTA_API_USER": {
      "required": true
    },
    "TWILIO_ACCOUNT_SID": {
      "required": true
    },
    "TWILIO_AUTH_TOKEN": {
      "required": true
    },
    "TWILIO_NUMBER": {
      "required": true
    }
  },
  "formation": {
    "web": {
      "quantity": 1
    }
  },
  "addons": [

  ],
  "buildpacks": [
    {
      "url": "heroku/nodejs"
    }
  ]
}
