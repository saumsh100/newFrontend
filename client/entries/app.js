
import '../apps/default';
import '../apps/Dashboard';
