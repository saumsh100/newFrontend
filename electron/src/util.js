
module.exports.getChannelFromVersion = (version) => {
  const splitVersion = version.split('-');
  return splitVersion[1] || null;
};
