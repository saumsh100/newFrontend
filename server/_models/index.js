
const Sequelize = require('sequelize');
const { postgres } = require('../config/globals');

const sequelizeConfig = {
  logging: false,
  dialect: 'postgres',
  host: postgres.host,
  port: postgres.port,
  define: {
    underscored: false,
    paranoid: true,
  },
};

// Don't use NODE_ENV===production
// It doesn't let us debug production locally easily
if (postgres.ssl) {
  sequelizeConfig.dialectOptions = {
    ssl: true,
  };
}

// If true, sequelize will dump all PostgreSQL queries into terminal
if (postgres.logging) {
  sequelizeConfig.logging = console.log // eslint-disable-line
}

// initialize sequelize
const sequelize = new Sequelize(
  postgres.database,
  postgres.username,
  postgres.password,
  sequelizeConfig,
  );

// Test the connection for logs
sequelize
  .authenticate()
  .then(() => {
    console.log(`Sequelize connected to database=${postgres.database} at host:port=${postgres.host}:${postgres.port}`);
  })
  .catch(err => {
    console.error('postgres config', postgres);
    console.error('Unable to connect to the database:', err);
  });

// Import and store all models.
const models = [];
// models.push((require('./Address').default(sequelize, Sequelize)));
models.push((require('./Account').default(sequelize, Sequelize)));
models.push((require('./Appointment').default(sequelize, Sequelize)));
models.push((require('./AuthSession').default(sequelize, Sequelize)));
models.push((require('./Call').default(sequelize, Sequelize)));
models.push((require('./Chair').default(sequelize, Sequelize)));
models.push((require('./Chat').default(sequelize, Sequelize)));
models.push((require('./Enterprise').default(sequelize, Sequelize)));
models.push((require('./Family').default(sequelize, Sequelize)));
models.push((require('./Invite').default(sequelize, Sequelize)));
models.push((require('./Patient').default(sequelize, Sequelize)));
models.push((require('./PatientUser').default(sequelize, Sequelize)));
models.push((require('./Permission').default(sequelize, Sequelize)));
models.push((require('./PinCode').default(sequelize, Sequelize)));
models.push((require('./Practitioner').default(sequelize, Sequelize)));
models.push((require('./Practitioner_Service').default(sequelize, Sequelize)));
models.push((require('./PractitionerRecurringTimeOff').default(sequelize, Sequelize)));
models.push((require('./Recall').default(sequelize, Sequelize)));
models.push((require('./Reminder').default(sequelize, Sequelize)));
models.push((require('./Request').default(sequelize, Sequelize)));
models.push((require('./Segment').default(sequelize, Sequelize)));
models.push((require('./SentRecall').default(sequelize, Sequelize)));
models.push((require('./SentReminder').default(sequelize, Sequelize)));
models.push((require('./Service').default(sequelize, Sequelize)));
models.push((require('./SyncClientError').default(sequelize, Sequelize)));
models.push((require('./SyncClientVersion').default(sequelize, Sequelize)));
models.push((require('./TextMessage').default(sequelize, Sequelize)));
models.push((require('./Token').default(sequelize, Sequelize)));
models.push((require('./User').default(sequelize, Sequelize)));
models.push((require('./WaitSpot').default(sequelize, Sequelize)));
models.push((require('./WeeklySchedule').default(sequelize, Sequelize)));

const db = {};
models.forEach((model) => {
  db[model.name] = model;
});

Object.keys(db).forEach((modelName) => {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

module.exports = db;
