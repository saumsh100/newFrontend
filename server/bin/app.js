
require('../config/initializeCodeTranspiler');

const globals = require('../config/globals');
const applyWebpack = require('../config/webpack/applyWebpack');
const handleErrors = require('../middleware/handleErrors');
const chalk = require('chalk');
const app = require('../config/express');

// require('../config/kue');

// Uses the NODE_ENV to determine logging full stack or not
app.set('showStackError', true);

// Load in webpack configurations
applyWebpack(app);


// Set the Handlebars templating engine
app.set('views', `${globals.root}/views`);
app.engine('hbs', require('../config/handlebars').engine);
app.set('view engine', 'hbs');

// Extra logging for communication with server
app.use((req, res, next) => {
  console.log(chalk.blue(req.method, req.originalUrl));
  next();
});


// Bind the routes

app.use(require('../routes'));

// Catch errors, log and respond to client
app.use(handleErrors);

// assume 404 since no middleware responded
/*app.use((req, res) => {
  res.status(404).render('404', {
    url: req.originalUrl,
    error: 'Not found',
  });
});*/

module.exports = app;
