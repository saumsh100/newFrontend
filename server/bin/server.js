
const http = require('http');
const globals = require('../config/globals');
const app = require('./app');
const Availability = require('../models/Availability');
const Patient = require('../models/Patient');

const server = http.createServer(app);

const io = require('socket.io')(server);

// TODO: implement Thinky!!!
io.on('connection', (socket) => {
  console.log('a new socket connected');
  socket.on('fetchAvailabilities', () => {
    Availability.run()
      .then((availabilities) => {
        socket.emit('receiveAvailabilities', availabilities);
      });
  });
  
  socket.on('fetchPatients', () => {
    Patient.run()
      .then((patients) => {
        socket.emit('receivePatients', patients);
      });
  });
  
  socket.on('addAvailability', (data) => {
    new Availability(data).save().then((availability) => {
      io.sockets.emit('availabilityAdded', availability);
    });
  });
  
  socket.on('removeAvailability', (data) => {
    Availability.get(data.id).then((availability) => {
      availability.delete()
        .then((result) => {
          io.sockets.emit('availabilityRemoved', result);
        });
    });
  });
  
  socket.on('syncClientTrigger', (data) => {
    console.log('sync client triggers!', 'data:', data);
  });
});

server.listen(globals.port, () => {
  console.log(`CareCru HTTP Server is running on port ${globals.port}`);
});
