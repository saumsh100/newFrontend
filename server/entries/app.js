
import '../../client/apps/default';
import '../../client/apps/Dashboard';
