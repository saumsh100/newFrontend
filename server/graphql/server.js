
import { Router } from 'express';
import graphQLHTTP from 'express-graphql';
import { sequelizeAuthMiddleware } from '../middleware/auth';
import schema from './data/schema';

const GraphQLRouter = Router();

const PORT = 4000;

const GraphQLServer = graphQLHTTP({
  schema,
  graphiql: true,
  pretty: true,
  subscriptionsEndpoint: `ws://localhost:${PORT}/subscriptions`,
});

// Reusing auth middleware as the server is exposed the same way an API would.
GraphQLRouter.all('*', sequelizeAuthMiddleware, GraphQLServer);

export default GraphQLRouter;

