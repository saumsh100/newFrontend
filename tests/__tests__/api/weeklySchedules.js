
import request from 'supertest';
import app from '../../../server/bin/app';
import generateToken from '../../util/generateToken';
import { WeeklySchedule } from '../../../server/_models';
import wipeModel from '../../util/wipeModel';
import { weeklyScheduleId, seedTestWeeklySchedules, officeHourId } from '../../util/seedTestWeeklySchedules';
import { seedTestUsers, wipeTestUsers } from '../../util/seedTestUsers';
import { omitPropertiesFromBody } from '../../util/selectors';
import filterObject from './chats';

const rootUrl = '/api/weeklySchedules';
const id = '8ce3ba61-60cd-40c6-bc85-c018cabd4a44';

function omitProperties(body) {
  filterObject(body, 'updatedAt');
  filterObject(body, 'createdAt');
  filterObject(body, 'id');
  return omitPropertiesFromBody(
    body,
    ['mondayId', 'tuesdayId', 'wednesdayId', 'thursdayId', 'fridayId', 'saturdayId', 'sundayId'],
  );
}

describe('/api/weeklySchedules', () => {
  // Seed with some standard user data
  let token = null;
  beforeEach(async () => {
    await seedTestUsers();
    await wipeModel(WeeklySchedule);
    await seedTestWeeklySchedules();
    token = await generateToken({
      username: 'manager@test.com',
      password: '!@CityOfBudaTest#$',
    });
  });

  afterAll(async () => {
    await wipeTestUsers();
    await wipeModel(WeeklySchedule);
  });

  describe('POST /', async () => {
    test('should create a weekly schedule and have the appropriate defaults', () => request(app)
      .post(rootUrl)
      .set('Authorization', `Bearer ${token}`)
      .send({ id })
      .expect(201)
      .then(({ body }) => {
        expect(omitProperties(body)).toMatchSnapshot();
      }));

    test('should create a weekly schedule and use customized daily schedule if provided', () => request(app)
      .post(rootUrl)
      .set('Authorization', `Bearer ${token}`)
      .send({
        id,
        monday: {
          startTime: '1996-08-02T16:00:00.000Z',
          endTime: '1996-08-03T16:00:00.000Z',
        },
      })
      .expect(201)
      .then(({ body }) => {
        expect(omitProperties(body)).toMatchSnapshot();
      }));

    test('should create a weekly schedule with monday closed if provided startTime equals to endTime', () =>
      request(app).post(rootUrl)
        .set('Authorization', `Bearer ${token}`)
        .send({
          id,
          monday: {
            startTime: '1996-08-02T16:00:00.000Z',
            endTime: '1996-08-02T16:00:00.000Z',
          },
        })
        .expect(201)
        .then(({ body }) => {
          expect(omitProperties(body)).toMatchSnapshot();
        }));
  });

  // TODO: Diff debug
  /*
  describe('GET /', () => {
    beforeEach(async () => {
      await seedTestWeeklySchedule();
    });

    test('retrieve weekly schedules', () => {
      return request(app)
        .get('/api/weeklySchedules')
        .set('Authorization', `Bearer ${token}`)
        .send({
          accountId,
        })
        .expect(200)
        .then(({ body }) => {
          expect(body).toMatchSnapshot();
        });
    });
  });
  */

  describe('PUT /:weeklyScheduleId', () => {
    test('should update a weekly schedule', () => request(app)
      .put(`${rootUrl}/${weeklyScheduleId}`)
      .set('Authorization', `Bearer ${token}`)
      .send({ isAdvanced: true })
      .expect(200)
      .then(({ body }) => {
        body = omitPropertiesFromBody(body);
        expect(body).toMatchSnapshot();
      }));

    test('should update monday dailySchedule for weeklySchedule if provided', () => request(app)
      .put(`${rootUrl}/${officeHourId}`)
      .set('Authorization', `Bearer ${token}`)
      .send({
        monday: {
          startTime: '2008-01-31T16:00:00.000Z',
          endTime: '2008-02-01T01:00:00.000Z',
        },
      })
      .expect(200)
      .then(({ body }) => {
        body = omitPropertiesFromBody(body);
        filterObject(body, 'updatedAt');
        filterObject(body, 'createdAt');
        expect(body).toMatchSnapshot();
      }));

    test('should update monday and set the startTime equal to endTime for weeklySchedule if isClosed for monday is set to be true', () =>
      request(app)
        .put(`${rootUrl}/${officeHourId}`)
        .set('Authorization', `Bearer ${token}`)
        .send({ monday: { isClosed: true } })
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          filterObject(body, 'updatedAt');
          filterObject(body, 'createdAt');
          expect(body).toMatchSnapshot();
        }));
  });
});
