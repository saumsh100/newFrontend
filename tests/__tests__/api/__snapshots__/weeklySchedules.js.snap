// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/weeklySchedules POST / create a weekly schedule 1`] = `
Object {
  "entities": Object {
    "weeklySchedules": Object {
      "2bd862e8-67ce-4328-8b2a-7df838ddbeea": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "friday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "id": "2bd862e8-67ce-4328-8b2a-7df838ddbeea",
        "monday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "saturday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "sunday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "thursday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "tuesday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "wednesday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
      },
    },
  },
  "result": "2bd862e8-67ce-4328-8b2a-7df838ddbeea",
}
`;

exports[`/api/weeklySchedules PUT /:weeklyScheduleId update a weekly schedule 1`] = `
Object {
  "entities": Object {
    "weeklySchedules": Object {
      "8ce3ba61-60cd-40c6-bc85-c018cabd4a40": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "friday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "id": "8ce3ba61-60cd-40c6-bc85-c018cabd4a40",
        "isAdvanced": false,
        "monday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "saturday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "sunday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "thursday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "tuesday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
        "wednesday": Object {
          "endTime": "1970-02-01T01:00:00.000Z",
          "isClosed": false,
          "startTime": "1970-01-31T16:00:00.000Z",
        },
      },
    },
  },
  "result": "8ce3ba61-60cd-40c6-bc85-c018cabd4a40",
}
`;
